src/wlserver.cpp

```diff
@@ -931,6 +1082,29 @@ void wlserver_touchdown( double x, double y, int touch_id, uint32_t time )
 
 		wlserver.mouse_surface_cursorx = tx;
 		wlserver.mouse_surface_cursory = ty;
+	break;
+
+		default: /* we are using the "auto" enum case to ensure compatibility for devices that were already using this*/
+		case DRM_MODE_PANEL_ORIENTATION_AUTO:
+
+		tx = g_bRotated ? y : x;
+		ty = g_bRotated ? 1.0 - x : y;
+		
+		tx *= g_nOutputWidth;
+		ty *= g_nOutputHeight;
+
+		tx += focusedWindowOffsetX;
+		ty += focusedWindowOffsetY;
+
+		tx *= focusedWindowScaleX;
+		ty *= focusedWindowScaleY;
+
+		wlserver.mouse_surface_cursorx = tx;
+		wlserver.mouse_surface_cursory = ty;
```

Comment:
This logic is pretty repetitive, isn't there a way to simplify it?

Reply:
Not entirely sure. but I'll try.

Reply:
Reworked with the help of @ruineka, should only change initial values now.

Reply:
This looks much better, thanks!